src/password-field.tsx

```diff
@@ -39,6 +43,7 @@ const PasswordAdornment: React.FC<{
     };
     return (
         <IconButton
+            dataAttributes={{testid: 'visibility-button'}}
             aria-label={texts.togglePasswordVisibilityLabel}
```

Comment:
instead of using a data-testid, I'd use the aria-label to find this button in tests